### PR TITLE
Add Hallowe'en

### PR DESCRIPTION
**Pull request by Slayer366**
# New Port for Hallowe'en

## Game Information
- **Title**: Hallowe'en
- **URL**: [LinApple](https://github.com/Slayer366/linapple)

### CFW Tests
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
- [x] 480x320
- [x] 640x480
- [ ] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

**Comment by Slayer366** [comment]
Whoops!  I wasn't paying attention when I dropped that marquee in the parent dir instead of the portdir where I should have had it.  Should be good now 👍

File: ports/hallowe_en/Hallowe'en (Apple II).sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+get_controls
+
+GAMEDIR="/$directory/ports/hallowe_en"
+
+export LD_LIBRARY_PATH="$GAMEDIR/libs.${DEVICE_ARCH}:$LD_LIBRARY_PATH"
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+export TEXTINPUTPRESET="Name"
+export TEXTINPUTINTERACTIVE="Y"
+
+$ESUDO chmod 777 -R $GAMEDIR/*
```
Comment (Cebion) [inline ports/hallowe_en/Hallowe'en (Apple II).sh:27]: Is this really needed? Upon installing the port via pm permissions are set for the whole folder.

File: ports/hallowe_en/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+|X|Selects 1 Player at main screen|
+|Y|Starts game after selecting 1 Player|
+
+Press Down to zap holes in the floors.  Only certain parts of the floors can be zapped.
+
+## Compile
+
+Running this with sdl12-compat is recommended even with some systems that may not require it for best results.
+
+git clone https://github.com/Slayer366/linapple
```
Comment (Cebion) [inline ports/hallowe_en/README.md:23]: This should also be in the code block?

File: ports/hallowe_en/gameinfo.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <path>./Hallowe'en (Apple II).sh</path>
+    <name>Hallowe'en (Apple II)</name>
+    <desc>The player controls a magician that must collect a number of green pumpkins from a series of platforms. The character can move left and right, and can jump over enemies. They are also able to shoot a magical bolt which can erode a hole in the platform's floor blocking enemies. The player must climb up a number of ladders, and must avoid enemies including: spiders, choppers, bolts, skulls, bolts, and duplicators.</desc>
+    <releasedate>19830101T000000</releasedate>
+    <rating>0.60</rating>
+    <developer>Bernd Rass, Thomas Schumann</developer>
+    <publisher>MicroSparc, Inc.</publisher>
+    <genre>Action</genre>
+    <image>./hallowe_en/screenshot.png</image>
+		<marquee>./hallowe_en/marquee.png</marquee>
```
Comment (Cebion) [inline ports/hallowe_en/gameinfo.xml:13]: I'm not sure this is supported

File: ports/hallowe_en/hallowe_en/libs.aarch64/libSDL_image-1.2.so.0
Comment (Cebion) [inline ports/hallowe_en/hallowe_en/libs.aarch64/libSDL_image-1.2.so.0:1]: Have you followed my suggestion in using my newly compiled sdl_image1.2? The list of libraries seems excessive to me.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/hallowe_en/hallowe_en/libs.aarch64/libSDL_image-1.2.so.0
Comment (Slayer366) [reply ports/hallowe_en/hallowe_en/libs.aarch64/libSDL_image-1.2.so.0:1]: I'll hit you up in Discord regarding this.  I'd love to reduce the number of libs used, but the few attempts I've made have resulted in a non-functioning app.  I am all for doing this if it's possible.

Comment (Slayer366) [reply ports/hallowe_en/hallowe_en/libs.aarch64/libSDL_image-1.2.so.0:1]: Incredible!  Thanks to you, I've significantly reduced the number of required libs across all 4 major CFWs.  I went from needing 18 to only needing 8.  Bloody amazing!

**Review by Cebion — APPROVED** [review]

**Pull request merged**